frameworks/javascript/cw-2.js

```diff
@@ -255,196 +239,160 @@ try {
     // Results would be confusing because the user submitted code is compiled into a script where
     // additional code is injected and line numbers will not match.
     trace: function(ex) {
-      return (ex.stack || ex.toString() || '')
+      return (ex.stack || ex.toString() || "")
         .toString()
         // remove file names (ie: (/cli-runner/...))
-        .replace(/\s\(.*\)/g, '')
+        .replace(/\s\(.*\)/g, "")
         // remove at [eval] statements
-        .replace(/(at)*( Object.)*\s*[(]?\[eval\].*(:\d*)*[)]?\n/g, '')
+        .replace(/(at)*( Object.)*\s*[(]?\[eval\].*(:\d*)*[)]?\n/g, "")
         // remove stack trace beyond the Module information
-        .replace(/at Module[\w\s.:\d\n]*/g, '')
+        .replace(/at Module[\w\s.:\d\n]*/g, "")
         // remove at Object.<anonymous>
-        .replace(/\t*at Object.<\w*>\n/g, '')
+        .replace(/\t*at Object.<\w*>\n/g, "")
         // handleError is used to wrap top level code, so lets remove it so that it doesn't
         // confuse users who won't understand why it is there.
-        .replace('at Object.Test.handleError', '');
-    },
-    pass: function() {
-      _expect(true);
-    },
-    fail: function(message) {
-      _expect(false, message);
-    },
-    expect: function(passed, message, options) {
-      _expect(passed, message, options);
+        .replace("at Object.Test.handleError", "");
     },
+    pass: () => _expect(true),
+    fail: message => _expect(false, message),
+    expect: (passed, failMsg, options) => _expect(passed, failMsg, options);
     assertSimilar: function(actual, expected, msg, options) {
       this.assertEquals(Test.inspect(actual), Test.inspect(expected), msg, options);
     },
     assertNotSimilar: function(actual, expected, msg, options) {
       this.assertNotEquals(Test.inspect(actual), Test.inspect(expected), msg, options);
     },
-    assertEquals: function(actual, expected, msg, options) {
-      if (typeof(msg) == 'object') {
+    assertEquals: function(actual, expected, msg, options = {successMsg: "Value == " + Test.inspect(expected)}) {
```

Comment:
We can't have default parameters because we still have to support old NodeJS versions.